### PR TITLE
sidecar: Add args prometheus.get_config_interval and prometheus.timeout_get_config

### PR DESCRIPTION
**Pull request by vzemtsov**
Signed-off-by: Zemtsov Vladimir <vl.zemtsov@gmail.com>

## Changes

Add 2 args for Thanos sidecar:
`prometheus.get_config_interval` - How often the Prometheus config. Default - 30s
`prometheus.timeout_get_config` -Timeout for get Prometheus config. Default - 5s

Add docs for this changes

This needed, bc:
1. HardCode - it's bad
2. If u have really big Prometheus installation, u don't want make many config-requests

### CODE REVIEW

**Comment by yeya24** [comment]
> If u have really big Prometheus installation, u don't want make many config-requests

The config request is pretty lightweight to Prometheus I guess. Have you seen a performance issue on this? Or do you have any other use cases?

**Comment by vzemtsov** [comment]
Simular issue - https://github.com/thanos-io/thanos/issues/4121

In My installation Prometheus I have more then 1000 ServiceMonitors, and very big Prometheus installation (many shards).

Sometimes prometheus answered on requests `/api/v1/status/config` more then 5s

File: cmd/thanos/config.go [hunk 1]
```diff
@@ -75,6 +77,12 @@ func (pc *prometheusConfig) registerFlag(cmd extkingpin.FlagClause) *prometheusC
 	cmd.Flag("prometheus.ready_timeout",
 		"Maximum time to wait for the Prometheus instance to start up").
 		Default("10m").DurationVar(&pc.readyTimeout)
+	cmd.Flag("prometheus.get_config_interval",
+		"How often the Prometheus config").
+		Default("30s").DurationVar(&pc.getConfigInterval)
+	cmd.Flag("prometheus.timeout_get_config",
```
Comment (yeya24) [inline cmd/thanos/config.go:83]: rename it to `prometheus.get_config_timeout`

File: cmd/thanos/config.go [hunk 2]
```diff
@@ -75,6 +77,12 @@ func (pc *prometheusConfig) registerFlag(cmd extkingpin.FlagClause) *prometheusC
...
+		"Timeout for get Prometheus config").
```
Comment (yeya24) [inline cmd/thanos/config.go:84]: `for getting`

File: cmd/thanos/config.go [hunk 3]
```diff
@@ -75,6 +77,12 @@ func (pc *prometheusConfig) registerFlag(cmd extkingpin.FlagClause) *prometheusC
 	cmd.Flag("prometheus.ready_timeout",
 		"Maximum time to wait for the Prometheus instance to start up").
 		Default("10m").DurationVar(&pc.readyTimeout)
+	cmd.Flag("prometheus.get_config_interval",
+		"How often the Prometheus config").
```
Comment (yeya24) [inline cmd/thanos/config.go:81]: How often to get Prometheus config

File: docs/components/sidecar.md [hunk 4]
```diff
@@ -136,6 +138,8 @@ Flags:
       --prometheus.ready_timeout=10m
                                  Maximum time to wait for the Prometheus
                                  instance to start up
+      --prometheus.timeout_get_config=5s
+                                 Timeout for get Prometheus config
```
Comment (yeya24) [inline docs/components/sidecar.md:142]: Please regenerate the docs by `make docs`.

**Review by yeya24 — COMMENTED** [review]
Also please add a changelog as well.

**Comment by vzemtsov** [comment]
@yeya24 Hi
Need add something else?

**Comment by yeya24** [comment]
Please rebase main and update the change log then we can merge. Thanks!

**Comment by vzemtsov** [comment]
Done

**Review by yeya24 — APPROVED** [review]
LGTM

**Comment by GiedriusS** [comment]
(1) is not always bad, we are trying hard (and it's a really hard problem) to not have many different options because then it becomes really confusing (: but if this solves a practical problem then it looks good!

**Pull request merged**